crates/router/src/connector/dlocal.rs

```diff
@@ -69,18 +70,18 @@ where
         .change_context(errors::ConnectorError::RequestEncodingFailed)
         .attach_printable("Failed to sign the message")?;
         let auth_string: String = format!("V2-HMAC-SHA256, Signature: {}", encode(authz));
-        let headers = vec![
-            (headers::AUTHORIZATION.to_string(), auth_string),
-            (headers::X_LOGIN.to_string(), auth.x_login),
-            (headers::X_TRANS_KEY.to_string(), auth.x_trans_key),
-            (headers::X_VERSION.to_string(), "2.1".to_string()),
-            (headers::X_DATE.to_string(), date),
+        let headers: Vec<(String, masking::Secret<String, masking::ApiKey>)> = vec![
+            (headers::AUTHORIZATION.to_string(), auth_string.into()),
+            (headers::X_LOGIN.to_string(), auth.x_login.into()),
+            (headers::X_TRANS_KEY.to_string(), auth.x_trans_key.into()),
+            (headers::X_VERSION.to_string(), "2.1".to_string().into()),
+            (headers::X_DATE.to_string(), date.into()),
             (
                 headers::CONTENT_TYPE.to_string(),
-                Self.get_content_type().to_string(),
+                Self.get_content_type().to_string().into(),
             ),
         ];
-        Ok(headers)
+        Ok(concat_headers(headers, Vec::<(String, String)>::new()))
```

Comment:
Directly return `headers` instead? Follow that for every other place you're concatinating with an empty `Vec`.

Reply:
If we try to directly return the header, the transformation between, `impl HeaderValue -> Box<dyn HeaderValue>` isn't apparent, and the compiler cannot abstract it. For this to happen we need have a `.map` to perform `Box::new`. I'll create a new function for this. Thanks!